Sources/MongoSwift/BSON/CodableNumber.swift

```diff
@@ -0,0 +1,87 @@
+import Foundation
+
+/// A protocol for numbers that require encoding/decoding support but are not necessarily BSON types.
+internal protocol CodableNumber {
```

Comment:
as I mentioned in person the other day, it seems like "primitive" is not really the best way to describe a protocol that only contains number types, so hopefully this is more clear. I guess it's internal anyway so naming is not crucial

Reply:
👍